sdks/go/pkg/beam/core/runtime/graphx/serialize.go

```diff
@@ -43,14 +43,14 @@ func EncodeMultiEdge(edge *graph.MultiEdge) (*v1.MultiEdge, error) {
 	if edge.DoFn != nil {
 		ref, err := encodeFn((*graph.Fn)(edge.DoFn))
 		if err != nil {
-			return nil, fmt.Errorf("encode: bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to encode multi edge %v, bad userfn: %v", *edge, err)
```

Comment:
I'm a bit torn on including "multi edge" since the part that's important is that the user's DoFn isn't able to be encoded, while "multi edge" is an internal detail a pipeline author doesn't run into. My inclination would be to drop "multi edge" and simply say 
("failed to encode userfn %v:", *edge, err)

Including the *edge as context is good since it should help narrow things down for the reading user. Let me know if you want help triggering this message.

Here and the similar messages below.

Reply:
Done.